### PR TITLE
Strip down helper.sh

### PR DESCRIPTION
**Pull request by birdsarah**
Naive removal of all the parts that we're not using.

### CODE REVIEW

**Comment by birdsarah** [comment]
@bakulf, I'm happy to comment this PR so you can understand the decisions.

File: src/platforms/linux/daemon/helper.sh [hunk 1]
```diff
@@ -218,173 +147,66 @@ add_default() {
...
 	for i in "${ADDRESSES[@]}"; do
 		add_addr "$i"
 	done
 	set_mtu_up
 	set_dns
+	# Get the allowed ips from wg show (added to peer using set_conf always 0.0.0.0/0 ::/0)
+	# Then process them with add_default
 	for i in $(while read -r _ i; do for i in $i; do [[ $i =~ ^[0-9a-z:.]+/[0-9]+$ ]] && echo "$i"; done; done < <(wg show "$INTERFACE" allowed-ips) | sort -nr -k 2 -t /); do
-		add_route "$i"
+		add_default "$i"
```
Comment (bakulf) [inline src/platforms/linux/daemon/helper.sh:190]: we don't use 0.0.0./0 ::0. Actually it's rare we do it. If captive portal and/or local-area-network-access are enabled, we do use add_route.

**Review by bakulf — CHANGES_REQUESTED** [review]
Huge cleanup! Good! But restore the add_route. It's required for some of the under-development features.

File: linux/daemon/helper.sh [hunk 2]
```diff
@@ -189,25 +140,16 @@ get_fwmark() {
 }
 
 remove_firewall() {
-	if type -p nft >/dev/null; then
```
Comment (birdsarah) [inline linux/daemon/helper.sh:192]: @bakulf just to confirm i'm removing nft support.

Comment (bakulf) [reply linux/daemon/helper.sh:192]: ok!

File: linux/daemon/helper.sh [hunk 3]
```diff
@@ -218,173 +160,65 @@ add_default() {
...
-		cmd nft -f <(echo -n "$nftcmd")
-	else
-		echo -n "$restore" | cmd $iptables-restore -n
-	fi
+	ip $proto route add "$1" dev "$INTERFACE" table $table
+	ip $proto rule add not fwmark $table table $table
+	ip $proto rule add table main suppress_prefixlength 0
+
+    # Things work without this, but it looks important
```
Comment (bakulf) [inline linux/daemon/helper.sh:171]: indentation

File: linux/daemon/helper.sh [hunk 4]
```diff
@@ -16,26 +16,11 @@ export PATH="${SELF%/*}:$PATH"
...
-PRE_UP=( )
-POST_UP=( )
-PRE_DOWN=( )
-POST_DOWN=( )
-SAVE_CONFIG=0
 CONFIG_FILE=""
-PROGRAM="${0##*/}"
-ARGS=( "$@" )
-
-cmd() {
```
Comment (bakulf) [inline linux/daemon/helper.sh:32]: I kind of like cmd() because it helps us to debug the sequence of commands

**Review by bakulf — APPROVED** [review]

**Pull request merged**